browser/ui/webui/settings/brave_tor_snowflake_extension_browsertest.cc

```diff
@@ -0,0 +1,162 @@
+        ->GetInstalledExtension(kSnowflakeExtensionId);
+  }
+
+  void SetTorDisabledPolicy(bool value) {
+    policy::PolicyMap policies;
+    policies.Set(policy::key::kTorDisabled, policy::POLICY_LEVEL_MANDATORY,
+                 policy::POLICY_SCOPE_MACHINE, policy::POLICY_SOURCE_PLATFORM,
+                 base::Value(value), nullptr);
+    provider_.UpdateChromePolicy(policies);
+  }
+
+ private:
+  void SetUpInProcessBrowserTestFixture() override {
+    EXPECT_CALL(provider_, IsInitializationComplete(testing::_))
+        .WillRepeatedly(testing::Return(true));
+    policy::BrowserPolicyConnector::SetPolicyProviderForTesting(&provider_);
+  }
+
+  testing::NiceMock<policy::MockConfigurationPolicyProvider> provider_;
+};
+
+IN_PROC_BROWSER_TEST_F(TorSnowflakeExtensionBrowserTest, InstallFail) {
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(),
+                                           GURL("brave://settings/privacy")));
+  auto* web_contents = browser()->tab_strip_model()->GetActiveWebContents();
+  content::WebContentsConsoleObserver console_observer(web_contents);
+  console_observer.SetPattern("Could not fetch data from the Chrome Web Store");
+  ClickSnowflakeToggle(web_contents);
+  console_observer.Wait();
+  EXPECT_FALSE(IsSnowflakeToggled(web_contents));
+  EXPECT_TRUE(IsSnowflakeToggleEnabled(web_contents));
+}
+
+IN_PROC_BROWSER_TEST_F(TorSnowflakeExtensionBrowserTest,
+                       InstallDisableEnableUninstall) {
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(),
+                                           GURL("brave://settings/privacy")));
+  auto* web_contents = browser()->tab_strip_model()->GetActiveWebContents();
+  EXPECT_FALSE(IsSnowflakeToggled(web_contents));
+  EXPECT_TRUE(IsSnowflakeToggleEnabled(web_contents));
+
+  SimulateSnowflakeInstall();
+  EXPECT_TRUE(IsSnowflakeToggled(web_contents));
+  {
+    // toggle extension disabled->enabled->disabled
+    EnableSnowflake(false);
+    EXPECT_FALSE(IsSnowflakeToggled(web_contents));
+    EXPECT_TRUE(IsSnowflakeInstalled());
+
+    EnableSnowflake(true);
+    EXPECT_TRUE(IsSnowflakeToggled(web_contents));
+    EXPECT_TRUE(IsSnowflakeInstalled());
+
+    EnableSnowflake(false);
+  }
+
+  // enable from settings.
+  ClickSnowflakeToggle(web_contents);
+  EXPECT_TRUE(IsSnowflakeToggled(web_contents));
+  EXPECT_TRUE(IsSnowflakeInstalled());
+
+  // disabled from settings -> uninstalled.
+  ClickSnowflakeToggle(web_contents);
+  EXPECT_FALSE(IsSnowflakeToggled(web_contents));
+  EXPECT_FALSE(IsSnowflakeInstalled());
+}
+
+IN_PROC_BROWSER_TEST_F(TorSnowflakeExtensionBrowserTest, CheckPolicy) {
+  SetTorDisabledPolicy(true);
+
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(),
+                                           GURL("brave://settings/privacy")));
+  auto* web_contents = browser()->tab_strip_model()->GetActiveWebContents();
+  EXPECT_FALSE(IsSnowflakeToggled(web_contents));
+  EXPECT_FALSE(IsSnowflakeToggleEnabled(web_contents));
```

Comment:
Should we also test that the toggle is grayed out?

Reply:
IsSnowflakeToggleEnabled checks for grey out. Here it returns false that means it is grayed